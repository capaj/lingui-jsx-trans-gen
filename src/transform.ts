import { transformSync } from '@babel/core'

export const transform = (code: string, configOverloads) => {
  return transformSync(code, {
    presets: ['@babel/preset-typescript'],
    plugins: ['@babel/plugin-proposal-optional-chaining'],
    ...configOverloads
  })?.code
}
