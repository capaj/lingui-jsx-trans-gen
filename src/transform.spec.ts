import fs from 'fs'
import { transform } from './transform'

describe('transform', () => {
  it('should add a transform', async () => {
    const simple = fs.readFileSync('./test-files/simple.tsx', {
      encoding: 'utf8'
    })
    expect(
      transform(simple, {
        filename: 'simple.tsx'
      })
    ).toMatchSnapshot()
  })
})
