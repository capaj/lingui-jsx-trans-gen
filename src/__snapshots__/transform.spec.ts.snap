// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform should add a transform 1`] = `
"import React from 'react';
import { useWorkspacesQuery } from '../../../graphql/gqlPrivateSchema';
import Loader from '../../common/page/loader/Loader';
import { PageContentWrapper } from '../../common/page/Page';
import './YourWorkspaces.less';
export const YourWorkspaces = () => {
  const {
    data,
    loading
  } = useWorkspacesQuery();

  if (loading) {
    return <Loader />;
  }

  if (!data || (data === null || data === void 0 ? void 0 : data.workspaces.length) === 0) {
    return null;
  }

  return <PageContentWrapper className=\\"YourWorkspaces\\">
      <div className=\\"grid-header\\">
        <h3>Your workspaces {a}</h3>
      </div>

      <div>
        Read the <a href=\\"https://lingui.js.org\\">documentation</a>
        for more info.
      </div>
    </PageContentWrapper>;
};"
`;
